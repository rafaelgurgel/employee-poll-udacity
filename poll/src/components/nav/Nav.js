// src/components/Nav.js
import React from 'react'
import { Link, useNavigate } from 'react-router-dom'
import { useDispatch, useSelector } from 'react-redux'
import { logoutAuthedUser } from '../../actions/authedUser'
import getAvatar from '../../utils/getAvatar'

import './nav.css'


export default function Nav() {
  const dispatch = useDispatch()
  const authedUser = useSelector((state) => state.authedUser)
  const user = useSelector((state) => state.users[authedUser])
  const navigate = useNavigate()

  const handleLogout = () => {
    dispatch(logoutAuthedUser())
    navigate('/')
  }

  return (
    <nav className="nav">
      {authedUser && (
        <>
          <Link to="/home">Home</Link>
          <Link to="/add">New Poll</Link>
          <Link to="/leaderboard">Leaderboard</Link>

          <div className="nav-profile">
            <img
              src={user ? getAvatar(user.avatarURL) : 'https://via.placeholder.com/50'}
              alt={user ? `Avatar of ${user.name}` : 'Avatar'}
              className="nav-avatar"
            />
            <span>Hello, {user && user.name}</span>
            <button
              className="logout-button"
              onClick={handleLogout}
            >
              Logout
            </button>
          </div>
        </>
      )}
    </nav>
  )
}
