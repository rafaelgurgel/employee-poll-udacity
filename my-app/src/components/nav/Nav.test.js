import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import { BrowserRouter } from 'react-router-dom'
import Nav from './Nav'
import '@testing-library/jest-dom'

const mockReducer = (state = {
  authedUser: 'sarahedo',
  users: {
    sarahedo: { id: 'sarahedo', name: 'Sarah Edo', avatarURL: null },
  }
}) => state

const mockStore = createStore(mockReducer)

describe('Nav Component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <Provider store={mockStore}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
