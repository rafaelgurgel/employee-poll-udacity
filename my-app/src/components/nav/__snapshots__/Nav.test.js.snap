// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="nav"
  >
    <a
      href="/home"
    >
      Home
    </a>
    <a
      href="/add"
    >
      New Poll
    </a>
    <a
      href="/leaderboard"
    >
      Leaderboard
    </a>
    <div
      class="nav-profile"
    >
      <img
        alt="Avatar of Sarah Edo"
        class="nav-avatar"
        src="defaultAvatar.svg"
      />
      <span>
        Hello, Sarah Edo
      </span>
      <button
        class="logout-button"
      >
        Logout
      </button>
    </div>
  </nav>
</DocumentFragment>
`;
